root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -4892,6 +4892,14 @@ limited_link_type_combinations: [
              input_entity_type: 'artist',
     expected_relationship_type: 'otherdatabases',
             expected_clean_url: 'https://www.abc.net.au/triplejunearthed/artist/sophisticated-dingo/',
+},
+  // Tsutaya
+  {
+                     input_url: 'https://shop.tsutaya.co.jp/cd/product/4562494355418/',
+             input_entity_type: 'release',
+    expected_relationship_type: 'mailorder',
+            expected_clean_url: 'https://shop.tsutaya.co.jp/cd/product/4562494355418/',
+       only_valid_entity_types: ['release'],
```

Comment:
You are not actually restricting it to releases though :) If you want to do so (and it probably makes sense) you need to add a `validate` function that only returns `true` for `case LINK_TYPES.mailorder.release` - you can follow the one for YesAsia.

The other `mailorder` options are artist and label. If the site has pages for artists and/or labels, then we should probably allow those too :) If not, just don't validate them (YesAsia for example seems to have no label pages so we don't do a `case LINK_TYPES.mailorder.label` at all).

Reply:
Got it, thanks! Tsutaya does have artist pages, so I'll add a case for that as well.

Reply:
How does [this](https://github.com/metabrainz/musicbrainz-server/pull/2807/commits/a0989edd2862296b0b4f53a402dc03fa85e9d15b) look? The regex may be a bit funky, which is related to the difference between Tsutaya's URL format and those of YesAsia and Tower Records (both of whose validators I referenced in building the original commit and this one).

Reply:
It looks nice. There is only a small bug and possible improvements. See the other conversation https://github.com/metabrainz/musicbrainz-server/pull/2807#discussion_r1087706318 for details.

Reply:
Since all bugs and improvements related to cleanup and validation have been cleaned and tested, I'm going to "resolve" this for now.